src/pages/authenticator/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import OTPDisplay from 'components/Authenicator/OTPDisplay';
+import { getAuthCodes } from 'services/authenticator/authenticatorService';
+import { Button } from '@mui/material';
+import { CustomError } from 'utils/error';
+import { PAGES } from 'constants/pages';
+import { useRouter } from 'next/router';
+
+const OTPPage = () => {
+    const router = useRouter();
+    const [codes, setCodes] = useState([]);
+    const [searchTerm, setSearchTerm] = useState('');
+
+    useEffect(() => {
+        const fetchCodes = async () => {
+            try {
+                getAuthCodes()
+                    .then((res) => {
+                        setCodes(res);
+                    })
+                    .catch((err) => {
+                        if (err.message === CustomError.KEY_MISSING) {
+                            router.push({
+                                pathname: PAGES.CREDENTIALS,
+                                query: { redirectPage: PAGES.AUTHENICATOR },
+                            });
+                            return;
+                        }
+                        console.error('something wrong here', err);
```

Comment:
On mobile, we decided to not log the errors for auth, since the underlying error could bubble up sensitive data

Reply:
In the end, we ended up adding logs to debug and fix customer issues :D.
Can remove it if we want. But might need to back it back again in the future just in case of things need to be debugged.